lldb/tools/lldb-dap/src-ts/lldb-dap-server.ts

```diff
@@ -0,0 +1,130 @@
+import * as child_process from "node:child_process";
+import * as vscode from "vscode";
+
+function areArraysEqual<T>(lhs: T[], rhs: T[]): boolean {
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+  for (let i = 0; i < lhs.length; i++) {
+    if (lhs[i] !== rhs[i]) {
+      return false;
+    }
+  }
+  return true;
+}
```

Comment:
nit: Nodejs does include a utility for this https://nodejs.org/docs/latest/api/util.html#utilisdeepstrictequalval1-val2